_re2.cc

```diff
@@ -39,23 +39,19 @@ using std::nothrow;
 using re2::RE2;
 using re2::StringPiece;
 
+#include <structmember.h>
+
 
 typedef struct _RegexpObject2 {
   PyObject_HEAD
-  // __dict__.  Simpler than implementing getattr and possibly faster.
-  PyObject* attr_dict;
```

Comment:
Is this part of the change necessary or beneficial?  Regexp objects should be constructed very rarely in performance-sensitive programs.

Reply:
Not sure I got that. The dict stores `groups`/`groupindex` and `pattern` fields. I didn't change the way re2 objects are constructed/destroyed. 

However, based on your feedback I'll re-implement these fields for `RegexpObject2` (probably using getter/setter to avoid the performance penalties)

Reply:
Ah, I see what you mean. In general execution, you create the regex once and then just match against it! That's true but I think we can get some performance improvements out of this so, why not :)

Reply:
I think the default choice should be to leave working code as-is.  Let's try to keep this PR scoped to changes that have measurable benefits.  If you think changing RegexpObject2 is useful, let's handle that separately.